File: 0058-length-of-last-word/0058-length-of-last-word.cpp
class Solution {
public:
    int lengthOfLastWord(string s) {
        int size = s.size();
        int i = size - 1;
        int reqWordCount = 0;
        
        while (i>=0 && s[i] == ' ') {
                i--;
        }
        
        while (i>=0 && s[i] != ' ') {
            reqWordCount += 1;
            i--;
        }
        
        return reqWordCount;
        
    }
};

File: 0058-length-of-last-word/0058-length-of-last-word.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        
#         reqWord = ''
#         for i in range(len(s)-1, -1, -1):
#             if s[i] == ' ':
#                 return reqWord
#             reqWord = s[i] + reqWord
            
#         print(reqWord)

# Approach 1
#         reqWordCount = 0
#         for i in range(len(s)-1, -1, -1):
#             if s[i] == ' ' and reqWordCount != 0:
#                 return reqWordCount
#             elif s[i] == ' ':
#                 continue
#             else:
#                 reqWordCount += 1
                
#         return reqWordCount

# Approach using split()

        # wordBag = s.split()
        # return len(wordBag[-1])
        i = len(s) - 1
        reqWordCount = 0
        while i>=0 and s[i] == ' ':
            i -= 1
        while i>=0 and s[i] != ' ':
            reqWordCount += 1
            i -= 1
        
        return reqWordCount
    
    

